### ISSUE TITLE
String problem

### ISSUE THREAD

**Issue by bigbearbro** [RuedigerMoeller/fast-serialization#155]
today I got a problem when using fst.
I got a deserialized object containing a String field，the String field is not null but the char[]value of the string is null. so when I
use fastjson to write the object to jsonstring,a null pointer exception happened when encountering the method string.length() {return value.length}

**Comment by bigbearbro** [comment]
Step1,
I created a 
Class A {
String version1;
},
and then I use redis to store an instance of class A for an hour.

Step2,
I made an upgrade on Class A {
String version1,
String version2
}

when I took from the bytes[] stored in redis which is the 1st version of classA, and then deserialized it into the new version of classA. I expected the field version2 would be null, but it isn't, it has a reference, but the char[] value of it is null.

I was confused a lot.

jdk.1.7,linux,createDefaultConfiguration.

I made a study in the testcase yesterday, and I found a solution.

Test1, I used the method getObjectInput(InputStream). 
The issue was based on the following code:
// bais = new ByteArrayInputStream(bytes);
// FSTObjectInput in = conf.getObjectInput(bais);
// return (T) in.readObject(clazz);

Test2, when I changed it to getObjectInput(byte[]), the following code throws an ArrayOutBoundException. This is great!, although it didn't deserialize succussfully, but it told me there might be an version-version mistake
return (T) conf.getObjectInput(bytes).readObject();
